optaplanner-examples/src/main/java/org/optaplanner/examples/machinereassignment/solver/score/MachineReassignmentConstraintProvider.java

```diff
@@ -145,74 +158,121 @@ private Constraint transientUsage(ConstraintFactory factory) {
     /**
      * Load: The safety capacity for each resource for each machine should not be exceeded.
      */
-    private Constraint loadCost(ConstraintFactory factory) {
+    protected Constraint loadCost(ConstraintFactory factory) {
         return factory.from(MrMachineCapacity.class)
                 .join(MrProcessAssignment.class,
-                        equal(MrMachineCapacity::getMachine, MrProcessAssignment::getMachine)
-                )
+                        equal(MrMachineCapacity::getMachine, MrProcessAssignment::getMachine))
                 .groupBy((machineCapacity, processAssignment) -> machineCapacity,
-                        sumLong((machineCapacity, processAssignment) ->
-                                processAssignment.getUsage(machineCapacity.getResource()))
-                )
+                        sumLong((machineCapacity, processAssignment) -> processAssignment
+                                .getUsage(machineCapacity.getResource())))
                 .filter(((machineCapacity, usage) -> machineCapacity.getSafetyCapacity() < usage))
                 .penalizeLong(MrConstraints.LOAD_COST,
                         HardSoftLongScore.ONE_SOFT,
-                        (machineCapacity, usage) -> usage - machineCapacity.getSafetyCapacity());
+                        (machineCapacity, usage) -> machineCapacity.getResource().getLoadCostWeight()
+                                * (usage - machineCapacity.getSafetyCapacity()));
     }
 
     /**
      * availability(r) = capacity(m, r) - usage(m, r)
      * balanceCost = sum(max(0, multiplier * availability(m, r1) - availability(m, r2)))
      */
-    private Constraint balanceCost(ConstraintFactory factory) {
-        return factory.from(MrBalancePenalty.class)
-                .join(MrProcessAssignment.class)
-                .groupBy((penalty, processAssignment) -> penalty, (penalty, processAssignment) -> processAssignment.getMachine(),
-                        sumLong((penalty, processAssignment) -> processAssignment.getUsage(penalty.getOriginResource())),
-                        sumLong((penalty, processAssignment) -> processAssignment.getUsage(penalty.getTargetResource()))
-                )
-                .penalizeLong(MrConstraints.BALANCE_COST, HardSoftLongScore.ONE_SOFT, this::balanceCost);
+    protected Constraint balanceCost(ConstraintFactory factory) {
+        throw new UnsupportedOperationException("Not yet implemented.");
+        /*
```

Comment:
"Interesting" behavior of the eclipse formatter: the block comment has leading asterisk on each line. As a result, IDE cannot automatically uncomment such a block.

Reply:
Maybe that's fixable?